substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -221,39 +223,44 @@ pub mod pallet {
 		/// via rewards and slashes are not reported and thus need to be manually fixed by the
 		/// staker. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
 		///
-		/// Invariant: what comes out of this list will always be a nominator.
+		/// Invariant: what comes out of this list will always be an active nominator.
 		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
 
-		/// WIP: This is a noop as of now, the actual business logic that's described below is going
-		/// to be introduced in a follow-up PR.
+		/// Something that provides a sorted list of targets (aka electable and chilled
+		/// validators), used for NPoS election.
 		///
-		/// Something that provides a best-effort sorted list of targets aka electable validators,
-		/// used for NPoS election.
+		/// The changes to the approval stake of each validator are reported to this list through
+		/// [`Self::EventListeners`]. The target(s) approval stake in the list should be updated in
+		/// any of these cases:
+		/// 1. The stake of a validator or nominator is updated.
+		/// 2. A nominator or validator is chilled.
+		/// 3. The nominations of a voter are updated.
+		/// 4. A nominator or validator ledger is removed from the staking system.
 		///
-		/// The changes to the approval stake of each validator are reported to this. This means any
-		/// change to:
-		/// 1. The stake of any validator or nominator.
-		/// 2. The targets of any nominator
-		/// 3. The role of any staker (e.g. validator -> chilled, nominator -> validator, etc)
+		/// Unlike `VoterList`, the values in this list are always kept up to date with both the
+		/// voter and target ledger's state at all the time (even upon rewards, slashes, etc) and
+		/// thus it represent an accurate approval stake of all target accounts in the system.
 		///
-		/// Unlike `VoterList`, the values in this list are always kept up to date with reward and
-		/// slash as well, and thus represent the accurate approval stake of all account being
-		/// nominated by nominators.
+		/// Chilled validators will *not* be removed from this list. Even when chilled, the target's
+		/// approval voting must be kept up to date. In case a chilled validator re-validates their
+		/// intention to be a validator again, their target score is up to date with the nominations
+		/// in the system.
 		///
-		/// Note that while at the time of nomination, all targets are checked to be real
-		/// validators, they can chill at any point, and their approval stakes will still be
-		/// recorded. This implies that what comes out of iterating this list MIGHT NOT BE AN ACTIVE
-		/// VALIDATOR.
+		/// A target is removed from the list IFF its approval score is zero. This means that the
+		/// validator must be unbonded *AND* no staker is nominating it.
 		type TargetList: SortedListProvider<Self::AccountId, Score = BalanceOf<Self>>;
```

Comment:
is the size of this bounded?

Reply:
`SortedListProvider` is generally unbounded under the hood, but is meant to provide an efficient sorted `.iter` over the values. hope this helps? https://paritytech.github.io/polkadot-sdk/master/frame_election_provider_support/trait.SortedListProvider.html